src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -57,32 +57,37 @@ private OverdueFineCalculatorService(Clients clients) {
   public CompletableFuture<Result<LoanAndRelatedRecords>> createOverdueFineIfNecessary(
     LoanAndRelatedRecords records, WebContext context) {
 
+    Loan loan = records.getLoan();
+    if (loan == null || !loan.isOverdue(DateTime.now(DateTimeZone.UTC))) {
```

Comment:
If `ClockManager` is the problem, maybe we should use `loan.isOverdue()` here, and replace the call to `ClockManager` with `DateTime.now(DateTimeZone.UTC)` in method `Loan#isOverdue()` itself?

Reply:
I appreciate the investigation and I'm glad that these changes have produced a stable build. I think we may need to understand better why ClockManager is not working as intended in this part of the system.

The `ClockManager` is intended to produce a `DateTime` in UTC and the production code has been moved towards that rather than using `DateTime.now`. I would prefer we didn't move away from that without being able to confirm why that is the cause.

Reply:
Hi @alexanderkurash, it seems issue happens if the `OverdueFineCalculatorServiceTest` is being executed after `RequestsAPICreationTests`. It uses `api.support.APITests#mockClockManagerToReturnFixedDateTime` to mock **current** date time however it never resets the mocking.

I'd add following to `api.support.APITests#afterEach`:

```
    mockClockManagerToReturnDefaultDateTime();
```

Hope it helps.

Reply:
Thanks @bohdan-suprun, this seems to help. I've updated this pull request.

Reply:
Thanks @bohdan-suprun I prefer that approach, I totally agree the clock should be reset after every test.